### PR TITLE
[QI2-1087] Added E2E tests

### PR DESCRIPTION
**Pull request by NischalQuTech**

### CODE REVIEW

File: tests/e2e_test.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    qc.rx(math.pi / 2, 0)
+    qc.ry(math.pi / 2, 1)
+    qc.rz(math.pi / 2, 2)
+    qc.sdg(2)
+    qc.t(1)
+    qc.tdg(0)
+    # qc.reset(2)
+    qc.cz(1, 2)
+    qc.cp(math.pi / 2, 1, 2)
+    # qc.swap(0, 1)
```
Comment (Mythir) [inline tests/e2e_test.py:69]: You can remove the commented code

File: tests/e2e_test.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    _get_auth_tokens()
+    _run_e2e_tests(name=name)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Run E2E test on a backend.")
+    parser.add_argument(
+        "name",
+        type=str,
+        nargs="?",
```
Comment (Mythir) [inline tests/e2e_test.py:106]: why `nargs="?"`?

File: tests/e2e_test.py (hunk 1 above)
Comment (Mythir) [inline tests/e2e_test.py:69]: You can remove the commented code

File: qiskit_quantuminspire/api/settings.py [hunk 3]
```diff
@@ -66,3 +71,20 @@ def from_config_file(cls, path: Path = API_SETTINGS_FILE) -> ApiSettings:
 
         api_settings = path.read_text()
         return ApiSettings.model_validate_json(api_settings)
+
+    @staticmethod
+    async def _fetch_team_member_id(host: str, access_token: str) -> int:
```
Comment (Mythir) [inline qiskit_quantuminspire/api/settings.py:76]: Fetching and storing the team member id is only needed for the e2e test right? If yes, then this probably shouldn't be in the production code.

**Review by Mythir — CHANGES_REQUESTED** [review]

File: tests/e2e_test.py (hunk 2 above)
Comment (NischalQuTech) [reply tests/e2e_test.py]: removed

File: tests/e2e_test.py (hunk 1 above)
Comment (NischalQuTech) [reply tests/e2e_test.py]: Removed

File: qiskit_quantuminspire/api/settings.py (hunk 3 above)
Comment (NischalQuTech) [reply qiskit_quantuminspire/api/settings.py:75]: right, removed it

Comment (Mythir) [reply qiskit_quantuminspire/api/settings.py:75]: I still see the _fetch_team_member_id and get_team_member_id functions in this file

**Review by Mythir — CHANGES_REQUESTED** [review]

**Review by eliasKA — APPROVED** [review]

**Pull request merged**